### PR TITLE
ccache: 4.2.1 → 4.3

### PR DESCRIPTION
**Pull request by kira-bruneau**
###### Motivation for this change
Update to the latest version: https://github.com/ccache/ccache/releases/tag/v4.3

###### Things done
- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

@r-burns

### CODE REVIEW

**Comment by MetaDark** [comment]
Result of `nixpkgs-review pr 122492` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>ccache</li>
    <li>ccacheStdenv</li>
    <li>ccacheWrapper</li>
  </ul>
</details>

**Comment by MetaDark** [comment]
Result of `nixpkgs-review pr 122492` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>ccache</li>
    <li>ccacheStdenv</li>
    <li>ccacheWrapper</li>
  </ul>
</details>

File: pkgs/development/tools/misc/ccache/default.nix [hunk 1]
```diff
@@ -38,18 +37,21 @@ let ccache = stdenv.mkDerivation rec {
...
+    # test/run requires the compgen function which is available in
+    # bashInteractive, but not bash.
+    bashInteractive
+  ] ++ lib.optional stdenv.isDarwin xcodebuild;
+
   checkPhase = ''
-    export HOME=$(mktemp -d)
-    ctest --output-on-failure ${lib.optionalString stdenv.isDarwin ''
+    runHook preCheck
+    HOME=$(mktemp -d) ctest --output-on-failure ${lib.optionalString stdenv.isDarwin ''
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/ccache/default.nix:51]:
```suggestion
    export HOME=$(mktemp -d)
    ctest --output-on-failure ${lib.optionalString stdenv.isDarwin ''
```
just to be sure.

Comment (MetaDark) [reply pkgs/development/tools/misc/ccache/default.nix:51]: Ok, yep I can change it back.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 122492` at 3bd456e4 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>ccache</li>
    <li>ccacheStdenv</li>
  </ul>
</details>

**Review by r-burns — APPROVED** [review]
Changes LGTM 👍

**Pull request merged**